### PR TITLE
Feat integration tests

### PR DESCRIPTION
**Pull request by fazzatti**
See issue #151 for reference.

Changes:

- Add Stellar Test Ledger
- Implement minor utils for testing
- Fix channel accounts plugin bug when reading states from smart contracts
- Introduce integration tests
  - Each use case spins a pristine ledger to test
  - all use cases should enforce whole features and complete behaviors

### CODE REVIEW

File: src/jest.config.integration.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  transform: {
+    '^.+\\.tsx?$': [
+      'ts-jest',
+      {
+        tsconfig: 'tsconfig.json',
+        isolatedModules: true,
+      },
+    ],
+  },
+  transformIgnorePatterns: ['/node_modules/(?!ethereum-cryptography)'],
```
Comment (Brunonascdev) [inline src/jest.config.integration.js:30]: is this necessary?

File: src/jest.config.js [hunk 2]
```diff
@@ -17,4 +18,21 @@ module.exports = {
...
+      'ts-jest',
+      {
+        tsconfig: 'tsconfig.json',
+        isolatedModules: true,
+      },
+    ],
+  },
+  transformIgnorePatterns: ['/node_modules/(?!ethereum-cryptography)'],
+  moduleNameMapper: {
+    '\\.(css|less|sass|scss)$': 'identity-obj-proxy',
```
Comment (Brunonascdev) [inline src/jest.config.js:32]: don't seems to be necessary as well

File: src/setup-tests.ts [hunk 3]
```diff
@@ -2,5 +2,4 @@
 import * as matchers from 'jest-extended'
 expect.extend(matchers)
 
-jest.setTimeout(10000)
```
Comment (Brunonascdev) [inline src/setup-tests.ts:5]: removing the timeout can be dangerous; unless the CI already have a timeout.

File: src/tests/integration/channel-accounts-plugin/index.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+
+    beforeAll(async () => {
+      issuer = new DefaultAccountHandler({ networkConfig })
+      user = new DefaultAccountHandler({ networkConfig })
+      await issuer.initializeWithFriendbot()
+      await user.initializeWithFriendbot()
+      userExpectedBalance = 0
+    })
+
+    it('Can use the channel accounts helper tool to initiate 10 channels', async () => {
```
Comment (Brunonascdev) [inline src/tests/integration/channel-accounts-plugin/index.integration.test.ts:55]: awesome way to describe tests 🚀

**Review by Brunonascdev — APPROVED** [review]
looking awesome!

can't wait to use this test-ledger on other projects 🚀

File: src/jest.config.integration.js (hunk 1 above)
Comment (fazzatti) [reply src/jest.config.integration.js:30]: I broke down the tests into three separate configurations:

- use different timeouts when running just unit vs (integration or combined)
- use different parallelization when running just unit vs (integration or combined)
- have separate coverage outputs when running individual(unit or integration) or combined

In the end, the CI will consider the combined approach as the total coverage, while you can still run them individually and analyze separately the results

File: src/jest.config.js (hunk 2 above)
Comment (fazzatti) [reply src/jest.config.js:32]: same as the other:

```
I broke down the tests into three separate configurations:

- use different timeouts when running just unit vs (integration or combined)
- use different parallelization when running just unit vs (integration or combined)
- have separate coverage outputs when running individual(unit or integration) or combined

In the end, the CI will consider the combined approach as the total coverage, while you can still run them individually and analyze separately the results```

File: src/setup-tests.ts (hunk 3 above)
Comment (fazzatti) [reply src/setup-tests.ts:5]: I moved it to the separate config files so we can have different values depending on how you're executing it (just unit, just integration, or combined)

File: src/jest.config.js (hunk 2 above)
Comment (fazzatti) [reply src/jest.config.js:32]: Sorry, if you were referring to the ignore pattern precisely, it was necessary due to some internal imports from the cacti test tooling package :(

**Pull request merged**